lib/segment/src/index/field_index/geo_index.rs

```diff
@@ -481,7 +491,7 @@ impl GeoMapIndex {
 
         // Assume all selected points have `max_values_per_point` value hits.
         // Therefore number of points can't be less than `total_hits / max_values_per_point`
-        let min_hits_by_value_groups = sum / self.max_values_per_point();
+        let min_hits_by_value_groups = sum.checked_div(max_values_per_point).unwrap_or(0);
```

Comment:
you have already checked zero condition, do we need `checked_div` here? I find `unwrap_or(0)` confusing because i cannot say why `0`

Reply:
Smart! 

I added the `== 0` shortcut later so I forgot about this. I'll remove it since it is redundant.

Reply:
Done in 5cbc795f1840da4c933a0cf5554fc626c9e56197.

I'll merge when all is :green_circle: .